File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int sml = n>m?m:n;
        int big = n>m?n:m;
        int r;
        while(sml != 0){
            r = big%sml;
            big = sml;
            sml = r;
        }
        int gcd = big;
        int lcm = (n*m)/gcd;
        
        
        int[] answer = new int[2];
        answer[0] = gcd; answer[1] = lcm;
        return answer;
    }
    
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    const gcd = getGCD(n,m)
    const lcm = (n*m) / getGCD(n,m)
    return [gcd, lcm];
}

function getGCD(a,b){
    const min = Math.min(a,b)
    const max = Math.max(a,b)
    return min == 0 ? max : getGCD(min, max%min)
}